pyOCD/tools/gdb_server.py

```diff
@@ -62,9 +66,9 @@ def build_parser(self):
         parser.add_argument("--list-targets", action="store_true", dest="list_targets", default=False, help="List all available targets.")
         parser.add_argument("--json", action="store_true", dest="output_json", default=False, help="Output lists in JSON format. Only applies to --list and --list-targets.")
         parser.add_argument("-d", "--debug", dest="debug_level", choices=debug_levels, default='info', help="Set the level of system logging output. Supported choices are: " + ", ".join(debug_levels), metavar="LEVEL")
-        parser.add_argument("-t", "--target", dest="target_override", choices=supported_targets, default=None, help="Override target to debug.  Supported targets are: " + ", ".join(supported_targets), metavar="TARGET")
-        parser.add_argument("-n", "--nobreak", dest="break_at_hardfault", default=True, action="store_false", help="Disable halt at hardfault handler.")
-        parser.add_argument("-r", "--reset-break", dest="break_on_reset", default=False, action="store_true", help="Halt the target when reset.")
+        parser.add_argument("-t", "--target", dest="target_override", default=None, help="Override target to debug.", metavar="TARGET")
+        parser.add_argument("-n", "--nobreak", dest="break_at_hardfault", default=True, action="store_false", help="Disable halt at hardfault handler. (Deprecated)")
```

Comment:
Yes, hard fault catch is on by default.

I'm reworking things a bit so the options dict passed to GDBServer only contains the vector_catch option. The `--nobreak` and `--reset-break` options are handled up front to form a single vector catch string. Also changing the `--nobreak` logic so it matches its name.

I was thinking specifying both would work like most command line tools, where the lattermost in the args takes precedence. Making it an error can be problematic if you are building a command line from different places, and need to override. Such as a default command line set by something, where you're given a field for additional command line options. — But this requires deciding whether `--nobreak` applies only to hard fault or all.